pkg/server/middleware/request_limits.go

```diff
@@ -22,11 +27,15 @@ func (l RequestLimits) Wrap(next http.Handler) http.Handler {
 		reader := io.LimitReader(r.Body, int64(l.maxRequestBodySize)+1)
 		body, err := io.ReadAll(reader)
 		if err != nil {
-			http.Error(w, fmt.Sprintf("failed to read request body: %v", err), http.StatusInternalServerError)
+			msg := fmt.Sprintf("failed to read request body: %v", err)
+			_ = level.Warn(l.logger).Log("msg", msg)
```

Comment:
I think it's better if we create a span logger for the context, so we have more details and traceability:
```go
log, ctx := spanlogger.New(r.Context(), "middleware.RequestLimits.Wrap")
defer log.Span.Finish()
...
level.Warn(log).Log("msg", msg)
```